modules/sr/robot3/marker.py

```diff
@@ -329,7 +327,18 @@ def yaw_pitch_roll(self) -> ThreeTuple:
 
         Specifically intrinsic Tait-Bryan angles following the z-y'-x'' convention.
         """
-        roll, pitch, yaw = self._quaternion.to_euler()
+        return self._calculate_yaw_pitch_roll(self._quaternion)
+
+    def _calculate_yaw_pitch_roll(self, q: Quaternion) -> ThreeTuple:
+        """
+        Calculate intrinsic Tait-Bryan angles following the z-y'-x'' convention.
```

Comment:
Is this change expected to change the values meaningfully for all cases, or is this more about handling edge cases better?

Separately: is there a reference for what the calculation is for these?